fiftyone/utils/eval/segmentation.py

```diff
@@ -449,13 +458,20 @@ def __init__(
         eval_key,
         pixel_confusion_matrix,
         classes,
+        matches=None,
         missing=None,
         backend=None,
     ):
         pixel_confusion_matrix = np.asarray(pixel_confusion_matrix)
-        ytrue, ypred, weights = self._parse_confusion_matrix(
-            pixel_confusion_matrix, classes
-        )
+        ytrue_ids = None
+        ypred_ids = None
+
+        if matches is not None:
```

Comment:
@prernadh mmm this an appealing way to store the data because it means `_parse_confusion_matrix` doesn't need to be called when loading results 🧠 

Prior to this refactor and before you created https://github.com/voxel51/fiftyone/pull/5330, I had forked your initial implementation with dicts and implemented Model Evaluation panel callbacks in https://github.com/voxel51/fiftyone/pull/5331.

As I mentioned on that PR, there's a few implementation details we probably aught to benchmark to make sure the panel is as performant as possible, but I could definitely see us using the data model here 👍